files/src/main/java/com/fincity/saas/files/service/AbstractFilesResourceService.java

```diff
@@ -739,6 +745,287 @@ public Mono<FileDetail> create(String clientCode, String uri, FilePart fp, Strin
+								FilesMessageResourceService.FORBIDDEN_PATH, this.getResourceType(), resourcePath);
+
+					Path path = Paths.get(this.getBaseLocation(), clientCode, resourcePath);
+
+					return this.createOrGetPath(path, urlResourcePath, fp, fileName, ovr, true);
+				},
+
+				(hasPermission, relativePath) -> {
+					// taking as an array so can update the path in future because Path data types behave as final
+					Path[] targetPath = new Path[1];
+
+					// updating the target path according to override condition
+					updatePathAccToOverrideCondition(fp, targetPath, filePath, ovr, fileName);
+					
+					// just checking the fileTpe if it is an image or not. if not work here then put in fp==null && ovr
+					String fileType = null;
+					try {
+						fileType = Files.probeContentType(targetPath[0]);
+					} catch (IOException e2) {
+						return Mono.empty();
+					}
+
+					if (fileType == null || !fileType.startsWith("image/"))
+						return Mono.empty();
+					
+					return updateAndProcessImage(fp, targetPath, fileName, ovr, imageDetails, urlResourcePath, clientCode);
+
+				})
+				.switchIfEmpty(msgService.throwMessage(msg -> new GenericException(HttpStatus.BAD_REQUEST, msg),
+						FilesMessageResourceService.IMAGE_FILE_REQUIRED))
+				.contextWrite(Context.of(LogUtil.METHOD_NAME, "AbstractFilesResourceService.create"));
+	}
+	
+	private void updatePathAccToOverrideCondition(FilePart fp, Path[] targetPath, String filePath, boolean ovr, String fileName) {
+		if(targetPath[0]!=null) {
+			if (fp == null) {
+				if (ovr) {
+					targetPath[0] = Paths.get(this.getBaseLocation(),
+							this.resolvePathWithClientCode(filePath).getT1());
+				} else {
+					targetPath[0] = Paths.get(this.getBaseLocation(),
+							this.resolvePathWithClientCode(filePath).getT1());
+					
+					for (int i = 1; i <= 100; i++) {
+						int lastIndex = filePath.lastIndexOf(".");
+						String extension = filePath.substring(lastIndex);
+						Path newImagePath = targetPath[0].resolve(fileName + i + extension);
+						
+			            if (!Files.exists(newImagePath)) {
+			            	try {
+			                    Files.copy(targetPath[0], newImagePath, StandardCopyOption.REPLACE_EXISTING);
+			                    targetPath[0] = newImagePath;
+			                    break;
+			                    
+			                } catch (IOException e) {
+			                    Mono.error(e);
+			                }
+			            }
+			        }
+				}
+			}
+		}else {
+			Mono.empty();
+		}
+	}
+	
+	private Mono<FileDetail> updateAndProcessImage(FilePart fp, Path[] path, String fileName, boolean ovr,
+                                      ImageDetails imageDetails, String urlResourcePath, String clientCode){
+		return FlatMapUtil.flatMapMonoWithNull(
+
+				() -> {
+					if (fp != null) {
+						return fp.transferTo(path[0]);
+					} else {
+						return Mono.empty();
+					}
+				},
+
+				x -> {
+					return Mono.just(path[0].toFile());
+				},
+
+				(x, actualFile) -> {
```

Comment:
Move the code inside try and catch block to a separate method to reduce the compexity of this method